src/blockMonitor/blockProcessor.ts

```diff
@@ -1,27 +1,62 @@
 import { ethers } from "ethers";
 import { StartStopService, ApplicationError } from "../dataEntities";
 import { ReadOnlyBlockCache, BlockCache } from "./blockCache";
-import { IBlockStub } from "./blockStub";
+import { IBlockStub } from "../dataEntities";
+import { Block, HasTxHashes } from "../dataEntities/block";
+
+type BlockFactory<T> = (provider: ethers.providers.Provider) => (blockNumberOrHash: number | string) => Promise<T>;
+
+export const minimalBlockFactory = (provider: ethers.providers.Provider) => async (
+    blockNumberOrHash: string | number
+): Promise<IBlockStub & HasTxHashes> => {
+    const block = await provider.getBlock(blockNumberOrHash);
+    return {
+        hash: block.hash,
+        number: block.number,
+        parentHash: block.parentHash,
+        transactions: block.transactions
+    };
+};
+
+export const defaultBlockFactory = (provider: ethers.providers.Provider) => async (
+    blockNumberOrHash: string | number
+): Promise<Block> => {
+    const block = await provider.getBlock(blockNumberOrHash);
+    const logs = await provider.getLogs({
+        blockHash: block.hash
+        // TODO: filter interesting logs, instead of taking everything
```

Comment:
TODO: still here

Reply:
Removed, added a comment for future notice.